micrometer-spring-legacy/src/samples/java/io/micrometer/spring/samples/PrometheusSample.java

```diff
@@ -15,16 +15,38 @@
  */
 package io.micrometer.spring.samples;
 
+import io.micrometer.core.api.MeterFactory;
+import io.micrometer.core.api.MeterProvider;
+import io.micrometer.core.instrument.Counter;
 import io.micrometer.spring.export.prometheus.EnablePrometheusMetrics;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 
 @SpringBootApplication(scanBasePackages = "io.micrometer.spring.samples.components")
 @EnablePrometheusMetrics
 @EnableScheduling
+@RestController
 public class PrometheusSample {
+    private static final Counter wayTooEarly = MeterFactory.counter("way_early_greetings");
+    private final Counter tooEarly = MeterFactory.counter("early_greetings");
+
+    @Autowired
+    MeterProvider meterProvider;
+
     public static void main(String[] args) {
         SpringApplication.run(PrometheusSample.class, args);
     }
-}
\ No newline at end of file
+
+    @RequestMapping("/hello")
+    public String hello() {
+        wayTooEarly.increment();
+        tooEarly.increment();
+        MeterFactory.counter("greetings").increment();
+        meterProvider.counter("injected_greetings").increment();
```

Comment:
OK my fears had proven true. The static and instance counters (`wayTooEarly` and `tooEarly`) had been created before the true implementation was set in the `@PostConstruct`. I solved it by creating the `SwappableMeterRegistry` that iterates through its counters and points them to counters from the new registry when it is set.

If this approach seems good to you @jkschneider, I'll finish the remaining implementation and JavaDocs.

Also I intend to remove these from the `PrometheusSample` before merging in in an effort to keep them simple. However I did want to include some sort of sample to discuss around,